### PR TITLE
Add ns-path op

### PR DESCRIPTION
**Pull request by expez**
This op returns the file path to the file containing some ns on classpath.

The new op is meant to be the basis for cider-find-ns

### CODE REVIEW

File: src/cider/nrepl/middleware/ns.clj [hunk 1]
```diff
@@ -51,6 +58,43 @@
   (transport/send transport (response-for msg :ns-vars (ns-vars msg)))
   (transport/send transport (response-for msg :status :done)))
 
+(defn- jar-file?
+  "Returns true if file is a normal file with a .jar or .JAR extension."
```
Comment (bbatsov) [inline src/cider/nrepl/middleware/ns.clj]: Is someone really using .JAR?

File: src/cider/nrepl/middleware/ns.clj [hunk 2]
```diff
@@ -51,6 +58,43 @@
...
+  [f]
+  (let [file (io/file f)]
+    (and (.isFile file)
```
Comment (bbatsov) [inline src/cider/nrepl/middleware/ns.clj]: The fact there's nothing like this in the standard lib is killing me.

File: src/cider/nrepl/middleware/ns.clj (hunk 1 above)
Comment (expez) [inline src/cider/nrepl/middleware/ns.clj]: I stole this function from `tools.namespace`, where it's private, so I assume so :(

File: src/cider/nrepl/middleware/ns.clj [hunk 3]
```diff
@@ -51,6 +58,43 @@
...
+         (or (.endsWith (.getName file) ".jar")
```
Comment (bbatsov) [inline src/cider/nrepl/middleware/ns.clj]: Guess you can downcase the filename and drop the `or`.

File: src/cider/nrepl/middleware/ns.clj [hunk 4]
```diff
@@ -70,4 +115,8 @@
     "ns-vars"
     {:doc "Returns a sorted list of all vars in a namespace."
      :requires {"ns" "The namespace to browse"}
-     :returns {"status" "done"}}}}))
+     :returns {"status" "done"}}
+    "ns-path"
+    {:doc "Returns the path to the file containing ns"
```
Comment (bbatsov) [inline src/cider/nrepl/middleware/ns.clj]: Add a `.` at the end of the sentence.

**Comment by expez** [comment]
Refactored.  

It's really annoying that the tests take like 20 mins.  Any thoughts about dropping support for all these old clojure versions?  The clojure team takes regression really seriously so I can't think of any reasons why someone would stay at say clojure 1.5 instead of upgrading.

**Comment by bbatsov** [comment]
> Any thoughts about dropping support for all these old clojure versions? The clojure team takes regression really seriously so I can't think of any reasons why someone would stay at say clojure 1.5 instead of upgrading.

That's out of question. We got some negative feedback when I decided to drop support for 1.3 and 1.4 when many projects were still supporting them. There's also the fact that 1.5 is [widely used yet](https://cognitect.wufoo.com/reports/state-of-clojure-2014-results/). 

Unless there's some strong incentive to do so, I'd like us to support as many versions as possible. We'll probably drop 1.5 once 1.8 is out.

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**